File: socialuni-server/socialuni-user-sdk/src/main/java/com/socialuni/social/user/sdk/dao/DO/SocialuniUserFollowDO.java
package com.socialuni.social.user.sdk.dao.DO;

import com.socialuni.social.common.api.entity.SocialuniContentBaseDO;
import com.socialuni.social.common.api.entity.SocialuniUserContactBaseDO;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

@Data
@Entity
@Table(name = "s_user_follow",
        indexes = {
                @Index(columnList = "userId"),
                @Index(columnList = "beUserId"),
                @Index(columnList = "status"),
                @Index(columnList = "updateTime"),
        },
        uniqueConstraints = {
                //一个人只能关注另一个人一次
                @UniqueConstraint(columnNames = {"userId", "beUserId"}),
        }
)
@EqualsAndHashCode(callSuper = true)
public class SocialuniUserFollowDO extends SocialuniUserContactBaseDO {
}

File: socialuni-server/socialuni-follow-sdk/src/main/kotlin/com/socialuni/social/friend/dao/DO/SocialuniUserFollowDO.kt
package com.socialuni.social.friend.dao.DO

import com.socialuni.social.common.api.entity.SocialuniUserContactBaseDO
import lombok.Data
import lombok.EqualsAndHashCode
import javax.persistence.Entity
import javax.persistence.Index
import javax.persistence.Table
import javax.persistence.UniqueConstraint

@Data
@Entity
@Table(
    name = "s_user_follow",
    indexes = [Index(columnList = "userId"), Index(columnList = "beUserId"), Index(columnList = "status"), Index(
        columnList = "updateTime"
    )],
    uniqueConstraints = [UniqueConstraint(columnNames = ["userId", "beUserId"])]
)
@EqualsAndHashCode(callSuper = true)
class SocialuniUserFollowDO : SocialuniUserContactBaseDO()